ext/numo/narray/gen/tmpl/binary.c

```diff
@@ -39,13 +39,46 @@ static void
             is_aligned_step(s2,sizeof(dtype)) &&
             is_aligned_step(s3,sizeof(dtype)) ) {
             //<% end %>
-            for (i=0; i<n; i++) {
+
+            if (s2 == 0){ // Broadcasting from scalar value.
                 check_intdivzero(*(dtype*)p2);
-                *(dtype*)p3 = m_<%=name%>(*(dtype*)p1,*(dtype*)p2);
-                p1 += s1;
-                p2 += s2;
-                p3 += s3;
+                if (s1 == sizeof(dtype) &&
+                    s3 == sizeof(dtype) ) {
+                    if (p1 == p3) { // inplace case
+                        for (i=0; i<n; i++) {
+                            ((dtype*)p1)[i] = m_<%=name%>(((dtype*)p1)[i],*(dtype*)p2);
```

Comment:
`((dtype*)p1)[i]` is not always equivalent with  `(dtype*)p1; p1 += s1;`. Such cases are specially taken cared at

https://github.com/naitoh/numo-narray/blob/edc712136b8b1a66c30b79a77466703fecc99eaf/ext/numo/narray/gen/tmpl/binary.c#L28-L30

Reply:
Sorry, you had a check `s1 == sizeof(dtype)` and `s3 == sizeof(dtype)`. Forget.